openedx/core/djangoapps/content_libraries/collections/rest_api/v1/views.py

```diff
@@ -0,0 +1,181 @@
+
+    def _verify_and_fetch_library_collection(self, library_key, collection_id, user, permission) -> Collection | None:
+        """
+        Verify that the collection belongs to the library and the user has the correct permissions
+        """
+        try:
+            library_obj = api.require_permission_for_library_key(library_key, user, permission)
+        except api.ContentLibraryNotFound as exc:
+            raise Http404 from exc
+
+        collection = None
+        if library_obj.learning_package_id:
+            collection = authoring_api.get_learning_package_collections(
+                library_obj.learning_package_id
+            ).filter(id=collection_id).first()
+        return collection
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Retrieve the Content Library Collection
+        """
+        lib_key_str = kwargs.pop('lib_key_str', None)
+        if not lib_key_str:
+            raise Http404
+
+        pk = kwargs.pop("pk", None)
+        library_key = LibraryLocatorV2.from_string(lib_key_str)
+
+        # Check if user has permissions to view this collection by checking if
+        # user has permission to view the Content Library it belongs to
+        collection = self._verify_and_fetch_library_collection(
+            library_key, pk, request.user, permissions.CAN_VIEW_THIS_CONTENT_LIBRARY
+        )
+
+        if not collection:
+            raise Http404
+
+        serializer = self.get_serializer(collection)
+        return Response(serializer.data)
+
+    def list(self, request, *args, **kwargs):
+        """
+        List Collections that belong to Content Library
+        """
+        lib_key_str = kwargs.pop('lib_key_str', None)
+        if not lib_key_str:
+            raise Http404
+
+        # Check if user has permissions to view collections by checking if user
+        # has permission to view the Content Library they belong to
+        library_key = LibraryLocatorV2.from_string(lib_key_str)
+        try:
+            content_library = api.require_permission_for_library_key(
+                library_key, request.user, permissions.CAN_VIEW_THIS_CONTENT_LIBRARY
+            )
+        except api.ContentLibraryNotFound as exc:
+            raise Http404 from exc
+
+        collections = authoring_api.get_learning_package_collections(content_library.learning_package.id)
+        serializer = self.get_serializer(collections, many=True)
+        return Response(serializer.data)
+
+    def create(self, request, *args, **kwargs):
+        """
+        Create a Collection that belongs to a Content Library
+        """
+        lib_key_str = kwargs.pop('lib_key_str', None)
+        if not lib_key_str:
+            raise Http404
+
+        # Check if user has permissions to create a collection in the Content Library
+        # by checking if user has permission to edit the Content Library
+        library_key = LibraryLocatorV2.from_string(lib_key_str)
```

Comment:
Worth handling as part of `convert_exceptions` I think?

Reply:
Yes, having it handled as part of `convert_exceptions` would make sense since this issue also exists for all `/api/libraries/v2/[lib_key_str]` endpoints